File: c++/prog11.cpp
#include <iostream>
#include <string>
using namespace std;

class data{
    public:
    int a,b;
    void setdata(){
        int x,y;
        cout<<"enter the number x and y : ";
        cin>>x>>y;
        a=x;
        b=y;
    }
};

class maximum : public data{
public:
void process(data c,data d){
    if(c.a>d.b){
        cout<<"x is big "<<endl;
    }
    else{
        cout<<"y is big ";
    }

}
};
int main()
{
   
 maximum max;
 max.setdata();
 max.process(max,max);
    return 0;
}

File: java/prog11.java

import java.util.Scanner;

class Data {

    float num1, num2;

}

class maximum extends Data {

    public void max(float a, float b) {
        num1 = a;
        num2 = b;
        if (num1 > num2) {
            System.out.println("num1 is greater");
        } else if (num2 > num1) {
            System.out.println("num2 is greater");
        } else {
            System.out.println("both are equal");
        }
    }

}

public class prog11 {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        maximum A = new maximum();
        A.max(1, 2);
        sc.close();
    }
}
